sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayListenerImpl.java

```diff
@@ -248,17 +259,23 @@ public ApplicationGatewayListenerImpl withHttps() {
     @Override
     public ApplicationGatewayListenerImpl withHostname(String hostname) {
         this.innerModel().withHostname(null);
-        // future work: add hostname instead of replacing
-        if (hostname == null) {
-            this.innerModel().withHostNames(null);
-        } else {
+        if (hostname != null) {
             List<String> hostNames = new ArrayList<>();
             hostNames.add(hostname);
             this.innerModel().withHostNames(hostNames);
         }
         return this;
     }
 
+    @Override
+    public ApplicationGatewayListenerImpl withHostnames(List<String> hostnames) {
+        this.innerModel().withHostname(null);
```

Comment:
Move this line into `if (hostname != null) {`?

Reply:
Moved.